lib/httpoison.ex

```diff
@@ -55,6 +55,108 @@ defmodule HTTPoison.Request do
           params: params,
           options: options
         }
+
+  @doc """
+  Returns an equivalent `curl` command for the given request.
+
+  ## Examples
+      iex> request = %HTTPoison.Request{url: "https://api.github.com", method: :get, headers: [{"Content-Type", "application/json"}]}
+      iex> HTTPoison.Request.to_curl(request)
+      "curl -X GET -H 'Content-Type: application/json' https://api.github.com ;"
+
+      iex> request = HTTPoison.get!("https://api.github.com", [{"Content-Type", "application/json"}]).request
+      iex> HTTPoison.Request.to_curl(request)
+      "curl -X GET -H 'Content-Type: application/json' https://api.github.com ;"
+  """
+  @spec to_curl(t()) :: binary()
+  def to_curl(request = %__MODULE__{}) do
```

Comment:
I wonder if we should change this to be `{:ok, binary} | {:error, term}` and we return the reason of the error instead of returning an error message? 

`{:ok, "curl -X POST http://test.com"}` if it's possible to generate the curl command. 

`{:error, :multipart_not_supported}` for example could be returned for the case when multipart is selected.